cmd/silkworm.cpp

```diff
@@ -14,291 +14,109 @@
-    log_opts.add_flag("--log.nocolor", log_settings.log_nocolor, "Disable colors on log lines");
-    log_opts.add_flag("--log.utc", log_settings.log_utc, "Prints log timings in UTC");
-    log_opts.add_flag("--log.threads", log_settings.log_threads, "Prints thread ids");
-    log_opts.add_option("--log.file", log_settings.log_file, "Tee all log lines to given file name");
+using namespace silkworm;
 
-    cli.parse(argc, argv);
+static std::atomic<size_t> s_allocated_memory{0};
 
-    // Assign settings
-    node_settings.data_directory = std::make_unique<DataDirectory>(datadir, /*create=*/true);
-    node_settings.chaindata_env_config.max_size = parse_size(chaindata_max_size).value();
-    node_settings.chaindata_env_config.growth_size = parse_size(chaindata_growth_size).value();
-    if (node_settings.chaindata_env_config.growth_size > node_settings.chaindata_env_config.max_size / 2) {
-        throw std::invalid_argument("--chaindata.growthsize too wide");
-    }
+void* operator new(size_t size) {
+    s_allocated_memory += size;
+    return malloc(size);
+}
 
-    node_settings.batch_size = parse_size(batch_size).value();
-    node_settings.etl_buffer_size = parse_size(etl_buffer_size).value();
+void* operator new[](size_t size) {
+    s_allocated_memory += size;
+    return malloc(size);
+}
 
-    // Parse prune mode
-    std::optional<BlockNum> olderHistory, olderReceipts, olderTxIndex, olderCallTraces;
-    if (cli["--prune.h.older"]->count()) olderHistory.emplace(cli["--prune.h.older"]->as<BlockNum>());
-    if (cli["--prune.r.older"]->count()) olderReceipts.emplace(cli["--prune.r.older"]->as<BlockNum>());
-    if (cli["--prune.t.older"]->count()) olderTxIndex.emplace(cli["--prune.t.older"]->as<BlockNum>());
-    if (cli["--prune.c.older"]->count()) olderCallTraces.emplace(cli["--prune.c.older"]->as<BlockNum>());
+void operator delete(void* ptr, size_t size) noexcept {
+    s_allocated_memory -= size;
+    free(ptr);
+}
 
-    std::optional<BlockNum> beforeHistory, beforeReceipts, beforeTxIndex, beforeCallTraces;
-    if (cli["--prune.h.before"]->count()) beforeHistory.emplace(cli["--prune.h.before"]->as<BlockNum>());
-    if (cli["--prune.r.before"]->count()) beforeReceipts.emplace(cli["--prune.r.before"]->as<BlockNum>());
-    if (cli["--prune.t.before"]->count()) beforeTxIndex.emplace(cli["--prune.t.before"]->as<BlockNum>());
-    if (cli["--prune.c.before"]->count()) beforeCallTraces.emplace(cli["--prune.c.before"]->as<BlockNum>());
+void operator delete[](void* ptr, size_t size) noexcept {
+    s_allocated_memory -= size;
+    free(ptr);
+}
 
-    node_settings.prune_mode = db::parse_prune_mode(prune_mode,  //
-                                                    olderHistory, olderReceipts, olderTxIndex, olderCallTraces,
-                                                    beforeHistory, beforeReceipts, beforeTxIndex, beforeCallTraces);
+void operator delete(void* ptr) noexcept {
+    s_allocated_memory -= sizeof(ptr);
```

Comment:
This is not semantically correct. The memory released is the memory pointed to by ptr, not the ptr itself, so `s_allocated_memory -= sizeof(ptr);` is not correct.

Reply:
I honestly threw in some stuff there only to ensure build.
Without it, in fact, the project complains 
```
error: the program should also define ‘void operator delete [](void*)’ [-Werror=sized-deallocation]
   49 | void operator delete[](void* ptr, size_t size) noexcept {
      |      ^~~~~~~~
```
While for MSVC I could use `_msize` to get the segment size pointed by ptr I didn't pay much attention as, being this PR still a draft, I will polish knobs later on.

Reply:
Besides, to my knowledge, there is no direct way to know the size of a segment pointed by a raw ptr **unless you override memory allocations and prepend each segment with its own size**

Reply:
Removed by 1fef176